### PR TITLE
doc: add rfc about user specified compaction

### PR DESCRIPTION
**Pull request by levy5307**
### What problem does this PR solve? 
add rfc about user specified compaction

##### Tests 

- No code

### CODE REVIEW

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Add two classes named with `compaction_operation` and `compaction_rule`.
+
+`compaction_filter_rule` represents the compaction rule to filter the keys which are stored in rocksdb.
+There are three types of compaction filter rule:
+1. hashkey rule, which supports prefix match, postfix match and anywhere match.
+2. sortkey rule. Just like hashkey rule, it also supports prefix match, postfix match and anywhere match.
+3. ttl rule. It supports time range match with format [begin_ttl, end_ttl]
+
+`compaction_operation` represents the compaction operation. A compaction operation will be executed when all the corresponding compaction rules are matched.
+There are two types of compaction filter rule:
```
Comment (hycdong) [inline rfcs/2021-05-27-user-specified-compaction.md:18]: There are two types of compaction operation

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+In pegasus, sometimes we should add user specified compaction policy to reduce disk usage. This RFC proposes a user specified compaction design.
+
+## Design
+
+Add two classes named with `compaction_operation` and `compaction_rule`.
+
+`compaction_filter_rule` represents the compaction rule to filter the keys which are stored in rocksdb.
+There are three types of compaction filter rule:
+1. hashkey rule, which supports prefix match, postfix match and anywhere match.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:13]:
```suggestion
1. Hashkey rule, which supports prefix match, postfix match and anywhere match.
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+2. sortkey rule. Just like hashkey rule, it also supports prefix match, postfix match and anywhere match.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:14]:
```suggestion
2. Sortkey rule. Just like hashkey rule, it also supports prefix match, postfix match and anywhere match.
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+3. ttl rule. It supports time range match with format [begin_ttl, end_ttl]
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:15]:
```suggestion
3. TTL rule. It supports time range match with format [begin_ttl, end_ttl]
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+`compaction_operation` represents the compaction operation. A compaction operation will be executed when all the corresponding compaction rules are matched.
+There are two types of compaction filter rule:
+1. delete. It represents that we should delete this key when all the rules are matched.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:19]:
```suggestion
1. Delete. It represents that we should delete this key when all the rules are matched.
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+2. update ttl. It represents that we should update ttl when all the rules are matched.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:20]:
```suggestion
2. Update TTL. It represents that we should update TTL when all the rules are matched.
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+In Pegasus, sometimes we should add user specified compaction policy to reduce disk usage. This RFC proposes a user specified compaction design.
+
+## Design
+
+Add two classes named with `compaction_operation` and `compaction_rule`.
+
+`compaction_filter_rule` represents the compaction rule to filter the keys which are stored in rocksdb.
+There are three types of compaction filter rule:
+1. Hashkey rule, which supports prefix match, postfix match and anywhere match.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:13]:
```suggestion
- Hashkey rule, which supports prefix match, postfix match and anywhere match.
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+In Pegasus, sometimes we should add user specified compaction policy to reduce disk usage. This RFC proposes a user specified compaction design.
+
+## Design
+
+Add two classes named with `compaction_operation` and `compaction_rule`.
+
+`compaction_filter_rule` represents the compaction rule to filter the keys which are stored in rocksdb.
+There are three types of compaction filter rule:
+- Hashkey rule, which supports prefix match, postfix match and anywhere match.
+2. Sortkey rule. Just like hashkey rule, it also supports prefix match, postfix match and anywhere match.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:14]:
```suggestion
- Sortkey rule. Just like hashkey rule, it also supports prefix match, postfix match and anywhere match.
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+3. TTL rule. It supports time range match with format [begin_ttl, end_ttl]
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:15]:
```suggestion
- TTL rule. It supports time range match with format [begin_ttl, end_ttl]
```

File: rfcs/2021-05-27-user-specified-compaction.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+`compaction_operation` represents the compaction operation. A compaction operation will be executed when all the corresponding compaction rules are matched.
+There are two types of compaction filter rule:
+1. Delete. It represents that we should delete this key when all the rules are matched.
+2. Update TTL. It represents that we should update TTL when all the rules are matched.
```
Comment (acelyc111) [inline rfcs/2021-05-27-user-specified-compaction.md:20]:
```suggestion
- Delete. It represents that we should delete this key when all the rules are matched.
- Update TTL. It represents that we should update TTL when all the rules are matched.
```

**Review by acelyc111 — APPROVED** [review]
LGTM

**Review by hycdong — APPROVED** [review]

**Pull request merged**